symphony-bdk-core/src/main/java/com/symphony/bdk/core/extension/BdkApiClientFactoryAware.java

```diff
@@ -0,0 +1,14 @@
+package com.symphony.bdk.core.extension;
+
+import com.symphony.bdk.core.client.ApiClientFactory;
+
+import org.apiguardian.api.API;
+
+/**
+ * TODO javadoc
+ */
+@API(status = API.Status.EXPERIMENTAL)
+public interface BdkApiClientFactoryAware {
```

Comment:
What to you mean by existing object?

Reply:
something that the extension would need but that would not be managed by the BDK itself, let's say that the extension needs to encrypt data before sending it via the generated api classes
so it needs an Encryptor object but also want to share it across multiple extensions so we would like to inject it in different places (like the ApiFactory that we inject in the extension but for an object not created/managed by the BDK)

(yes I'm making this up and maybe this is too complex for now)

Reply:
This would be possible if the extension provides a service, this service could expose an `init()` methods that takes external objects not necessary managed by the BDK. Something like: 
```java
MyExtensionService service = bdk.extensions().service(MyExtension.class);
service.configure(encryptionModule);
// then
String encryptedData = service.encrypt("Hello, World!");
```